modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java

```diff
@@ -190,6 +201,17 @@ public void beforeNodeStop() {
             public boolean isStopped() {
                 return shutdownFuture.isDone();
             }
+
+            @Override
+            public void updateMetadata(NodeMetadata metadata) {
+                cluster.updateMetadata(metadata).subscribeOn(scheduler).subscribe();
+                MembershipEvent membershipEvent = createUpdated(cluster.member(),
+                        cluster.<NodeMetadata>metadata().map(METADATA_CODEC::serialize).orElse(null),
+                        METADATA_CODEC.serialize(metadata),
+                        System.currentTimeMillis()
+                );
+                topologyService.onMembershipEvent(membershipEvent);
```

Comment:
Why do we need to emit an event here? I can see that `updateMetadata` calls `updateIncarnation`, which will propagate this information to the remote peers

Reply:
It's a local event. We have the same logic when a cluster starts. See org.apache.ignite.network.AbstractClusterService#start

Reply:
why do we need a local event in this particular case?